### PR TITLE
[codex] Refactor ThinkTank around benches and native review runs

### PR DESCRIPTION
**Pull request by phrazzld**
## Summary

This PR refactors ThinkTank into a thin bench launcher for Pi-based research and review runs.

It replaces the older dispatch flow with bench resolution, YAML/config overlays, per-run artifacts, and first-party `research/default` and `review/cerberus` benches.

## What Changed

- replaced the CLI with `run`, `research`, `review`, and `benches` commands
- added typed bench, agent, provider, and run-contract structs
- introduced layered config loading from built-ins, user config, and optional trusted repo config
- launched all agent work through the Pi subprocess runner and recorded raw outputs plus optional synthesis artifacts
- added native review inputs for `--base`, `--head`, `--repo`, and `--pr`
- expanded tests around config resolution, bench execution, artifact persistence, and CLI edge cases

## Why

ThinkTank should stay a thin harness: it owns launch, isolation, concurrency, timeouts, and artifact recording while Pi agents own the reasoning. Benches are the right abstraction boundary here because they keep the interface simple without introducing a semantic workflow DSL.

## Impact

Users can now:

- run `thinktank research` as a first-class alias for the default research bench
- run `thinktank review` for local diff or PR-oriented multi-agent review
- define providers, agents, and benches declaratively in code or YAML
- consume stable artifact layouts for follow-on tooling and integrations

## Validation

- `mix format --check-formatted`
- `mix compile --warnings-as-errors`
- `mix credo`
- `MIX_ENV=dev mix dialyzer`
- `MIX_ENV=test mix coveralls`
- `mix hex.audit`
- `mix escript.build`
- `./thinktank --help`

### CODE REVIEW

File: lib/thinktank/cli.ex [hunk 1]
```diff
@@ -113,260 +146,353 @@ defmodule Thinktank.CLI do
...
+      action: :run,
+      bench_id: bench_id,
+      cwd: File.cwd!(),
+      json: parsed[:json] || false,
+      output: parsed[:output] && Path.expand(parsed[:output]),
+      dry_run: parsed[:dry_run] || false,
+      trust_repo_config: parsed[:trust_repo_config] || false,
+      input: %{
+        input_text: input_text,
+        paths: normalize_paths(parsed[:paths]),
```
[the remark this answers is not in the document]
Comment (moomooskycow) [reply lib/thinktank/cli.ex]: Fixed in 9fea90c. `build_common_command/3` now uses `Keyword.get_values/2` before normalization, so repeated `--paths` flags are preserved instead of dropping everything after the first value.

File: lib/thinktank/cli.ex [hunk 2]
```diff
@@ -113,260 +146,353 @@ defmodule Thinktank.CLI do
...
-    """
+  defp render_artifact_lines(artifacts) do
+    Enum.map_join(artifacts, "\n", fn artifact ->
+      "- #{artifact["name"]}: #{artifact["file"]}"
+    end)
+  end
+
+  defp resolve_bench(bench_id, parsed) do
+    with {:ok, config} <-
+           Config.load(cwd: File.cwd!(), trust_repo_config: parsed[:trust_repo_config] || false) do
```
[the remark this answers is not in the document]
Comment (moomooskycow) [reply lib/thinktank/cli.ex]: Fixed in 9fea90c. The CLI now preserves the tri-state `trust_repo_config` value and only passes that option when explicit, so `THINKTANK_TRUST_REPO_CONFIG` continues to work as the default.

File: lib/thinktank/executor/agentic.ex [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+    name
+    |> String.downcase()
+    |> String.replace(~r/[^a-z0-9]+/, "-")
+    |> String.trim("-")
+  end
+
+  defp validate_agent_config_dir!(base_dir) do
+    base_dir
+    |> Path.join("**")
+    |> Path.wildcard(match_dot: true)
```
[the remark this answers is not in the document]
Comment (moomooskycow) [reply lib/thinktank/executor/agentic.ex:293]: Fixed in 9fea90c. `validate_agent_config_dir!/1` now rejects a symlinked root before walking descendants, so trusted repo config cannot redirect `PI_CODING_AGENT_DIR` through the top-level path.

File: AGENTS.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+- Benches are named sets of agents
+- ThinkTank launches Pi, captures raw outputs, and records artifacts
+
+## What ThinkTank Is Not
+
+- Not a semantic workflow engine
+- Not a stage graph DSL
+- Not a prose parser
+- Not a second direct-API path that bypasses Pi
+
```
[the remark this answers is not in the document]
Comment (moomooskycow) [reply AGENTS.md:19]: The mismatch here was the stale PR metadata, not the repo guidance. This branch remains a thin bench launcher by design, so I updated the PR title and body to describe benches and native review runs instead of changing `AGENTS.md` toward a workflow-engine architecture that is not actually present.

File: CLAUDE.md [hunk 5]
```diff
@@ -2,85 +2,52 @@
...
+lib/thinktank/cli.ex                → CLI, aliases, dry-run, bench commands
+lib/thinktank/engine.ex             → bench resolution and launch orchestration
+lib/thinktank/builtin.ex            → built-in agents and benches
+lib/thinktank/config.ex             → built-in + user + repo config loading
+lib/thinktank/bench_spec.ex         → typed bench config
+lib/thinktank/agent_spec.ex         → typed agent config
+lib/thinktank/executor/agentic.ex   → Pi subprocess launcher
+lib/thinktank/run_store.ex          → raw artifacts and manifest
+lib/thinktank/run_contract.ex       → persisted run contract
 ```
```
[the remark this answers is not in the document]
Comment (moomooskycow) [reply CLAUDE.md:23]: Same disposition here: the implementation does not add a workflow engine or non-agentic executor, so `CLAUDE.md` is intentionally still bench-launcher oriented. I updated the PR title and body to match the shipped architecture, and kept the contributor guidance aligned with that actual design.

File: lib/thinktank.ex [hunk 6]
```diff
@@ -1,8 +1,5 @@
 defmodule Thinktank do
   @moduledoc """
-  Agent-first research tool for multi-perspective AI analysis.
-
-  Dispatches research questions to multiple LLM perspectives and
-  synthesizes their responses into coherent insight.
+  Thin Pi bench launcher for research and code review.
```
[the remark this answers is not in the document]
Comment (moomooskycow) [reply lib/thinktank.ex:3]: This moduledoc is still accurate for the code on this branch. The stale workflow-engine framing was in the PR description, so I corrected the PR metadata rather than rewriting the module docs toward an architecture the codebase does not implement.

File: lib/thinktank/cli.ex [hunk 7]
```diff
@@ -113,260 +146,353 @@ defmodule Thinktank.CLI do
...
+      json: parsed[:json] || false,
+      output: parsed[:output] && Path.expand(parsed[:output]),
+      dry_run: parsed[:dry_run] || false,
+      trust_repo_config: parsed[:trust_repo_config] || false,
+      input: %{
+        input_text: input_text,
+        paths: normalize_paths(parsed[:paths]),
+        agents: parse_agent_list(parsed[:agents]),
+        no_synthesis: parsed[:no_synthesis] || false
+      }
```
[the remark this answers is not in the document]
Comment (moomooskycow) [reply lib/thinktank/cli.ex:306]: This is not a bug in the current flow. `Engine.resolve/3` normalizes inputs through `normalize_input/2`, which stringifies incoming keys before `maybe_run_synthesizer/8` checks `%{"no_synthesis" => true}`, so CLI-provided atom keys are normalized before the skip branch runs.

File: lib/thinktank/config.ex [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+  end
+
+  defp build_benches(_), do: {:error, "benches must be a map"}
+
+  defp validate_references(benches, agents, providers) do
+    with :ok <- validate_agent_providers(agents, providers) do
+      Enum.reduce_while(benches, :ok, fn {_id, bench}, :ok ->
+        case validate_bench_references(bench, agents) do
+          :ok -> {:cont, :ok}
+          {:error, reason} -> {:halt, {:error, reason}}
```
[the remark this answers is not in the document]
Comment (moomooskycow) [reply lib/thinktank/config.ex]: Fixed in 9fea90c. Reference-validation errors now keep `bench.id`, and the helper flattening removed the Credo warning from the same code path.

File: AGENTS.md [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+- Agents are configured in code or YAML
+- Benches are named sets of agents
+- ThinkTank launches Pi, captures raw outputs, and records artifacts
+
+## What ThinkTank Is Not
+
+- Not a semantic workflow engine
+- Not a stage graph DSL
+- Not a prose parser
+- Not a second direct-API path that bypasses Pi
```
[the remark this answers is not in the document]
Comment (moomooskycow) [reply AGENTS.md:18]: This is the same stale-metadata issue as the earlier AGENTS thread. The code still intentionally avoids stage graphs and direct API execution paths, so I corrected the PR title and body to match the bench-launcher architecture instead of rewriting contributor guidance toward unsupported concepts.
